src/trunk.c

```diff
@@ -8811,75 +8824,135 @@ trunk_branch_count_num_tuples(trunk_handle *spl,
    }
 }
 
+/*
+ * Stats struct to return trunk node usage statistics via print interface
+ */
+typedef struct trunk_node_stats {
+   platform_log_handle *log_handle;
+   uint64               num_nodes;
+   uint64               num_tuples;
+   uint64               num_kv_bytes;
+} trunk_node_stats;
+
+/* Print summary branch information for given trunk node at address 'addr' */
 bool
 trunk_node_print_branches(trunk_handle *spl, uint64 addr, void *arg)
 {
-   platform_log_handle *log_handle = (platform_log_handle *)arg;
+   trunk_node_stats    *node_stats = arg;
+   platform_log_handle *log_handle = node_stats->log_handle;
    trunk_node           node;
    trunk_node_get(spl->cc, addr, &node);
 
+   // clang-format off
+   const char *dashes = "------------------------------------------------------------------";
+   const char *dashes_long = "-------------------------------------------------------------------------------------------------";
+   // clang-format on
+
+   platform_log(log_handle, "%s\n", dashes);
    platform_log(
-      log_handle,
-      "------------------------------------------------------------------\n");
-   platform_log(
-      log_handle, "| node %12lu height %u\n", addr, trunk_height(&node));
-   platform_log(
-      log_handle,
-      "------------------------------------------------------------------\n");
+      log_handle, "| Trunk node=%lu, height=%u\n", addr, trunk_height(&node));
+   platform_log(log_handle, "%s\n", dashes);
+
    uint16 num_pivot_keys = trunk_num_pivot_keys(spl, &node);
-   platform_log(log_handle, "| pivots:\n");
+   platform_log(log_handle, "| pivots: %u\n", num_pivot_keys);
    for (uint16 pivot_no = 0; pivot_no < num_pivot_keys; pivot_no++) {
       char key_str[128];
       trunk_key_to_string(spl, trunk_get_pivot(spl, &node, pivot_no), key_str);
       platform_log(log_handle, "| %u: %s\n", pivot_no, key_str);
    }
 
+   platform_log(log_handle, "%s\n", dashes_long);
+
    // clang-format off
    platform_log(log_handle,
-         "-----------------------------------------------------------------------------------\n");
-   platform_log(log_handle,
-         "| branch |     addr     |  num tuples  | num kv bytes |    space    |  space amp  |\n");
-   platform_log(log_handle,
-         "-----------------------------------------------------------------------------------\n");
+         "    | branch |     addr     |  num tuples  |       num kv bytes        |    space    |space amp |\n");
    // clang-format on
+
+   platform_log(log_handle, "%s\n", dashes_long);
+
    uint16 start_branch = trunk_start_branch(spl, &node);
    uint16 end_branch   = trunk_end_branch(spl, &node);
+   uint16 num_branches = 0;
+   uint64 sum_ntuples  = 0;
+   uint64 sum_nkvbytes = 0;
+
+   size_t nkvbytes = 0;
    for (uint16 branch_no = start_branch; branch_no != end_branch;
         branch_no        = trunk_add_branch_number(spl, branch_no, 1))
```

Comment:
@ajhconway -- We need to discuss how to re-work this iteration.

When I spoke with you briefly about this (in Dec 2022), you remarked that this iteration is possibly even incorrect, and may even have a race condition.

I recall you said something about this iteration is not paying attention to "dead pivots" [?] or something to that effect.

Would like to get some code references from `trunk.c` to places where we are doing the right iteration, so I can apply that logic here, too.